pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -40,6 +40,28 @@ func NewHelper() *Helper {
 func (h *Helper) InstallFlags(flags *pflag.FlagSet) {
 }
 
+// GetAllDockerAuths returns the entire docker config found, regardless of filtering
+// by imageName, and whether it was readfrom a file
+func (h *Helper) GetAllDockerAuths(authType string) (credentialprovider.DockerConfig, bool) {
+	log.V(3).Infof("Locating docker config for type %s", authType)
+	var searchPaths []string
+	var cfg credentialprovider.DockerConfig
+	var err error
+
+	if pathForAuthType := os.Getenv(authType); len(pathForAuthType) > 0 {
+		searchPaths = []string{pathForAuthType}
+	} else {
+		searchPaths = GetExtraSearchPaths()
+	}
+	log.V(3).Infof("Getting docker config in paths : %v", searchPaths)
+	cfg, err = GetDockerConfig(searchPaths)
+	if err != nil {
+		klog.Errorf("Reading docker config from %v failed: %v", searchPaths, err)
+		return credentialprovider.DockerConfig{}, false
+	}
+	return cfg, true
+}
+
 // GetDockerAuth returns a valid Docker AuthConfiguration entry, and whether it was read
```

Comment:
Correct it is no longer called.

I left it on on purpose in case it was needed in the future.  The use of the keyring stuff was foreign enough to me that I wanted it around for reference.

Reply:
Works for me.